pkg/controller/core/workload_controller.go

```diff
@@ -264,12 +325,53 @@ func (r *WorkloadReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
+// admittedNotReadyWorkload returns as pair of values. The first boolean determines
+// if the workload is currently counting towards the timeout for PodsReady, i.e.
+// it has the Admitted condition True and the PodsReady condition not equal
+// True (False or not set). The second value is the remaining time to exceed the
+// specified timeout counted since max of the LastTransitionTime's for the
+// Admitted and PodsReady conditions.
+func (r *WorkloadReconciler) admittedNotReadyWorkload(workload *kueue.Workload, clock clock.Clock) (bool, time.Duration) {
+	if r.podsReadyTimeout == nil {
+		// the timeout is not configured for the workload controller
+		return false, 0
+	}
+	if workload.Spec.Admission == nil {
+		// the workload is not admitted so there is no need to time it out
+		return false, 0
+	}
+	admittedCond := apimeta.FindStatusCondition(workload.Status.Conditions, kueue.WorkloadAdmitted)
+	if admittedCond == nil || admittedCond.Status != metav1.ConditionTrue {
+		// workload does not yet have the condition indicating its admission time
+		return false, 0
+	}
+	podsReadyCond := apimeta.FindStatusCondition(workload.Status.Conditions, kueue.WorkloadPodsReady)
+	if podsReadyCond != nil && podsReadyCond.Status == metav1.ConditionTrue {
+		return false, 0
+	}
+	elapsedTime := clock.Since(admittedCond.LastTransitionTime.Time)
+	if podsReadyCond != nil && podsReadyCond.Status == metav1.ConditionFalse && podsReadyCond.LastTransitionTime.After(admittedCond.LastTransitionTime.Time) {
+		elapsedTime = clock.Since(podsReadyCond.LastTransitionTime.Time)
+	}
+	waitFor := *r.podsReadyTimeout - elapsedTime
+	if waitFor < 0 {
+		waitFor = 0
+	}
+	return true, waitFor
+}
+
 func workloadStatus(w *kueue.Workload) string {
 	if apimeta.IsStatusConditionTrue(w.Status.Conditions, kueue.WorkloadFinished) {
 		return finished
 	}
 	if w.Spec.Admission != nil {
-		return admitted
+		if apimeta.IsStatusConditionTrue(w.Status.Conditions, kueue.WorkloadAdmitted) {
+			return admitted
+		}
+		return admitting
```

Comment:
We probably don't want this distinction, or it would slow down startup.

Reply:
It made the state transitions symmetric, and more granular. I think the same performance can be achieved if we reacted to `pending->admitting` as we reacted to the `pending->admitted`. However, it is not required for this PR, and may have some side effects if not done carefully, so reverting.

Reply:
Resolving as the change is reverted already